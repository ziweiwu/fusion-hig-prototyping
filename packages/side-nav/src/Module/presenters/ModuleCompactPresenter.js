import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { ThemeContext } from "@hig/themes";

import "./module.scss";

export default class ModuleCompact extends Component {
  static propTypes = {
    /** Indicates this module is currently active */
    active: PropTypes.bool,
    /** Indicates a nested submodule is currently active */
    activeChildren: PropTypes.bool,
    /** An instance of @hig/icon */
    icon: PropTypes.node.isRequired,
    /** Called when clicking on the title */
    onClickTitle: PropTypes.func,
    /** Called when hovering over the icon */
    onMouseOver: PropTypes.func
  };

  static defaultProps = {
    onClickTitle: () => {},
    onMouseOver: () => {}
  };

  render() {
    const {
      active,
      activeChildren,
      icon,
      onClickTitle,
      onMouseOver
    } = this.props;
    const classes = themeClass => cx(themeClass, "hig__side-nav__module");

    const linkClasses = themeClass =>
      cx(themeClass, "hig__side-nav__module__link", {
        "hig__side-nav__module__link--active": active,
        "hig__side-nav__module__link--active-children": activeChildren
      });

    return (
      <ThemeContext.Consumer>
        {({ themeClass }) => (
          <div className={classes(themeClass)} onMouseOver={onMouseOver}>
            <div className="hig__side-nav__module__row">
              <div
                className={linkClasses(themeClass)}
                onClick={onClickTitle}
                role="button"
                tabIndex={0}
              >
                <div className="hig__side-nav__module__link__icon">{icon}</div>
              </div>
            </div>
          </div>
        )}
      </ThemeContext.Consumer>
    );
  }
}
