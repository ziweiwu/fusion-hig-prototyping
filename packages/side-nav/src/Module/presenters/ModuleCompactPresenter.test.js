import React from "react";
import renderer from "react-test-renderer";
import Icon, { names as iconNames } from "@hig/icon";
import ModuleCompactPresenter from "./ModuleCompactPresenter";

describe("side-nav/Module/presenters/ModuleCompactPresenter", () => {
  describe("snapshot tests", () => {
    const cases = [
      {
        description: "renders with minimal props",
        props: {
          icon: <Icon name={iconNames.COLLABORATION} />
        }
      }
    ];

    cases.forEach(({ description, props }) => {
      it(description, () => {
        const wrapper = <ModuleCompactPresenter {...props} />;
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
