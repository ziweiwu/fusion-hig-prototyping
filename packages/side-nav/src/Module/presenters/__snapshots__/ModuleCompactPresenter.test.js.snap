// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/Module/presenters/ModuleCompactPresenter snapshot tests renders with minimal props 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={[Function]}
>
  <div
    className="hig__side-nav__module__row"
  >
    <div
      className="hig--light-theme hig__side-nav__module__link"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="hig__side-nav__module__link__icon"
      >
        <div
          className="hig__icon hig__icon--24-size"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" role=\\"img\\" title=\\"collaboration-24\\"><g fill=\\"#2e4258\\"><path d=\\"M8,9v6a1,1,0,0,0,1,1h7V9a1,1,0,0,0-1-1H8ZM9,9h6v6H9Z\\"/><path d=\\"M20,3H9A1,1,0,0,0,8,4V7H9V4H20V15H17v1h3a1,1,0,0,0,1-1V4A1,1,0,0,0,20,3Z\\"/><path d=\\"M15,17v3H4V9H7V8H4A1,1,0,0,0,3,9V20a1,1,0,0,0,1,1H15a1,1,0,0,0,1-1V17Z\\"/></g></svg>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
